### PR TITLE
Fix #1457 type in verse markers while typing.

### LINKED ISSUE #1457: Type in verse markers while typing
Tom has a team of typists. The typists take the handwritten work that was completed during a MAST event and type them into translationStudio. Their workflow seems to be interrupted and the team frustrated by having the verse markers reset and manually placing them in again and again after every edit. They would like to be able to just type the numbers in while typing the verse in and not having them reset later.

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Fix #1457 type in verse markers while typing.

Changes in this pull request:
Added check and message for duplicate verses.

@neutrinog - made latest fix - message for duplicate verses.

### CODE REVIEW

**Comment by neutrinog** [comment]
@PhotoNomad0 looks good except for one weird bug. Some times I'll create a few different verse markes and it won't render all of them. Or if I begin to edit it and then exit edit mode it will display the verse markers as plain text and add a new bubble at the beginning again.

On that note I think now we can completely disable rendering of the verse markers when in edit mode. Just let them edit them as raw USFM rather than displaying them as gray numbers.

**Comment by PhotoNomad0** [comment]
Thanks, will get to that this evening when I get home.

**Comment by PhotoNomad0** [comment]
@neutrinog - think I can recreate the problem you saw if there is not a space after the verse number.  I think we need to correct for that when leaving edit mode.

**Comment by PhotoNomad0** [comment]
@neutrinog- think I have this working now.  Please review.

**Comment by neutrinog** [comment]
@PhotoNomad0 now whenever I move between edit and drag mode the text gets duplicated. Also one time I marked it as done and some completely random text (from earlier in the history) was inserted.

File: app/src/main/java/com/door43/translationstudio/spannables/USFMVerseSpan.java [hunk 1]
```diff
@@ -147,4 +151,37 @@ public static USFMVerseSpan parseVerse(String usfm) {
             return new int[]{startVerse, endVerse};
         }
     }
+
+    /**
+     * fix verse markers missing the terminating space after verse number.
+     * @param in
+     * @return
+     */
+    public static CharSequence fixIncompleteVerseMarkers(CharSequence in) {
```
Comment (neutrinog) [inline app/src/main/java/com/door43/translationstudio/spannables/USFMVerseSpan.java]: note: I'm not certain we want to do this, but I don't think there's any need to change it for now.

Comment (PhotoNomad0) [inline app/src/main/java/com/door43/translationstudio/spannables/USFMVerseSpan.java]: sorry, impulsive there with the rush to deliver.  Just didn't want to hear a lot of complaining that verse marker was not recognized.

**Comment by PhotoNomad0** [comment]
@neutrinog - I think I know the problem.  I saw some broken code in the on done and tried to "fix" it.

**Comment by PhotoNomad0** [comment]
@neutrinog - have cleaned up the USFM verse marker regex and removed USFM verse marker cleanup code.  Made fix on done switch to make sure we are validating the latest text changes.

**Comment by neutrinog** [comment]
I added a change that also fixes https://github.com/unfoldingWord-dev/ts-android/issues/1369

**Pull request merged**